src/cclib/io/moldenwriter.py

```diff
@@ -0,0 +1,174 @@
+                                              x, y, z))
+
+        return block
+
+    def _gto_from_ccdata(self):
+        """Create [GTO] section using gbasis."""
+
+        # atom_sequence_number1 0
+        # shell_label number_of_primitives 1.00
+        # exponent_primitive_1 contraction_coeff_1 (contraction_coeff_1)
+        # ...
+        # empty line
+        # atom_sequence__number2 0
+        gbasis = self.ccdata.gbasis
+        label_template = '{:s} {:5d} 1.00'
+        basis_template = '{:15.6e} {:15.6e}'
+        block = []
+
+        for no, basis in enumerate(gbasis):
+            block.append('{:3d} 0'.format(no + 1))
+            for prims in basis:
+                block.append(label_template.format(prims[0].lower(),
+                                                   len(prims[1])))
+                for prim in prims[1]:
+                    block.append(basis_template.format(prim[0], prim[1]))
+            block.append('')
+
+        return block
+
+    def _scfconv_from_ccdata(self):
+        """Create [SCFCONV] section using gbasis."""
+
+        # scf-first    1 THROUGH   12
+        #    -672.634394
+        #    ...
+        #    -673.590571
+        #    -673.590571
+        block = ["scf-first    1 THROUGH   %d" % len(self.ccdata.scfenergies)]
+
+        for scfenergy in self.ccdata.scfenergies:
+            block.append('{:15.6f}'.format(scfenergy))
+
+        return block
+
+    def _mo_from_ccdata(self):
+        """Create [MO] section."""
+
+        # Sym= symmetry_label_1
+        # Ene= mo_energy_1
+        # Spin= (Alpha|Beta)
+        # Occup= mo_occupation_number_1
+        # ao_number_1 mo_coefficient_1
+        # ...
+        # ao_number_n mo_coefficient_n
+        # ...
+        moenergies = self.ccdata.moenergies
+        mocoeffs = self.ccdata.mocoeffs
+        homos = self.ccdata.homos
+        mult = self.ccdata.mult
+
+        has_syms = False
+        block = []
+
+        if hasattr(self.ccdata, 'mosyms'):
+            has_syms = True
+            syms = self.ccdata.mosyms[0]  # Optional
+
+        for i in range(mult):
+            spin = 'Alpha'
+            for j in range(len(moenergies)):
+                if has_syms:
+                    block.append(' Sym= %s' % syms[j])
+                moenergy = utils.convertor(moenergies[i][j], 'eV', 'hartree')
+                block.append(' Ene= {:10.4f}'.format(moenergy))
+                block.append(' Spin= %s' % spin)
+                if j <= homos[i]:
+                    block.append(' Occup= {:10.6f}'.format(2.0 / mult))
+                else:
+                    block.append(' Occup= {:10.6f}'.format(0.0))
+                for k, mocoeff in enumerate(mocoeffs[i][j]):
+                    block.append('{:4d}  {:10.6f}'.format(k + 1, mocoeff))
+
+            spin = 'Beta'
+
+        return block
+
+    def generate_repr(self):
+        """Generate the MOLDEN representation of the logfile data."""
+
+        molden_block = ['[Molden Format]']
```

Comment:
Having 'block' in the name is a little misleading, since this is a list of lines.